src/components/Modals/NewCourse.vue

```diff
@@ -1,24 +1,102 @@
+      <div v-if="!editMode" class="newCourse-requirements-container">
+        <div  class="newCourse-requirements" v-for="req in requirements" :key="req" :class="{'newCourse-space': !checkIfLast(req, requirements)}">
+        {{ checkIfLast(req, requirements) ? req : `${req},`}}
+        </div>
+      </div>
+      <div v-else class="newCourse-requirements-edit">
+        <editRequirement v-for="req in requirements" :key="req" :name="req" :selected="true"/>
+      </div>
+      <div class="newCourse-title">This class could potentially fulfill the following requirement(s):</div>
+      <div v-if="!editMode" class="newCourse-requirements-container">
+        <div class="newCourse-name" v-for="potreq in potentialReqs" :key="potreq" :class="{'newCourse-space': !checkIfLast(potreq, potentialReqs)}">
+          {{ checkIfLast(potreq, potentialReqs) ? potreq : `${potreq},`}}
+        </div>
+      </div>
+      <div v-else class="newCourse-requirements-edit">
+        <editRequirement v-for="potreq in potentialReqs" :key="potreq" :name="potreq" :isClickable="true"/>
+        <binaryButton  v-for="choice in binaryPotentialReqs" :key="choice[0]" :choices="choice"></binaryButton>
+      </div>
+      <div v-if="!editMode" class="newCourse-link" @click="toggleEditMode()">Add these Requirements</div>
+    </div>
   </div>
 </template>
 
 <script>
 import coursesJSON from '../../assets/courses/courses.json';
+import NewSemester from '@/components/Modals/NewSemester';
+import EditRequirement from '@/components/EditRequirement';
+import BinaryButton from '@/components/BinaryButton';
+
+Vue.component('newSemester', NewSemester);
+Vue.component('editRequirement', EditRequirement);
+Vue.component('binaryButton', BinaryButton);
+
+const fall = require('../../assets/images/fallEmoji.svg');
+const spring = require('../../assets/images/springEmoji.svg');
+const winter = require('../../assets/images/winterEmoji.svg');
+const summer = require('../../assets/images/summerEmoji.svg');
 
 export default {
+  components: { EditRequirement, BinaryButton },
   props: {
     isOnboard: Boolean,
     semesterID: Number,
-    placeholderText: String
+    placeholderText: String,
+    season: String,
+    year: Number,
+    goBack: Boolean
+  },
+  data() {
+    return {
+      seasonImg: {
+        Fall: fall,
+        Spring: spring,
+        Winter: winter,
+        Summer: summer
+      },
+      selected: false,
+      requirements: ['DummyReq1', 'DummyReq2'],
+      potentialReqs: ['PotentialReq1', 'PotentialReq2'],
+      binaryPotentialReqs: [['binarychoice1', 'binarychoice2']],
+      editMode: false,
+      selectedCourse: ''
+    };
+  },
+  watch: { 
+    goBack: function onPropChange(val) { // watch it
+      if (this.editMode) { this.editMode = false; }
+      else {
+        this.selected = false;
+        const impCopy = document.getElementById(`dropdown-${this.semesterID}`);
```

Comment:
Another thing we try to do is avoid using document.getElementById because with Vue we should be able to avoid this DOM Manipulation by passing data along and preserving state. If you copied this from somewhere else just let me know because I'm in the process of removing all of the old stuff like this we have - but if this is entirely new, can you do this in another way?

Reply:
Yeah I copied this from the `autocomplete()` function - in the same file and I think a similar technique is used for the NewSemester dropdown in `Modal.vue`

Reply:
I think im just going to leave it as is for now because there would be a lot of refactoring to do since it's used in many places

Reply:
Yup! I deleted a lot of this in my last PR but this will need to be deleted as well. Could you add a comment that you duplicated the code and where you did so here so I can refer to it later? https://courseplan.monday.com/boards/433956786/pulses/848072269